code/modules/bitrunning/virtual_domain/domains/beach_bar.dm

```diff
@@ -13,10 +13,18 @@
 	desc = "Whose drink is this? Not yours, that's for sure. Well, it's not like they're going to miss it."
 	list_reagents = list(/datum/reagent/consumable/ethanol/pina_colada = 30)
 
-/obj/item/reagent_containers/cup/glass/drinkingglass/filled/virtual_domain/Initialize(mapload, vol)
+/datum/lazy_template/virtual_domain/beach_bar/setup_domain(list/created_atoms)
 	. = ..()
 
-	AddComponent(/datum/component/bitrunning_points, \
-		signal_type = COMSIG_GLASS_DRANK, \
-		points_per_signal = 0.5, \
-	)
+	for(var/obj/item/reagent_containers/cup/glass/drink in created_atoms)
+		if(istype(drink, /obj/item/reagent_containers/cup/glass/drinkingglass/filled))
+			drink.list_reagents += list(/datum/reagent/consumable/ethanol/pina_colada = 30)
+			drink.add_initial_reagents()
```

Comment:
If these drinking glasses have already been initialized, then you shouldn't call `add_initial_reagents` a second time. 

You should just be doing `drink.reagents.add_reagent(/datum/reagent/consumable/ethanol/pina_colada, 30)`.